### DISCUSSION TITLE
System command issue

### DISCUSSION THREAD

**Discussion by pillouz01** [SilentVoid13/Templater#762] [Help]
Hi All,
I'm quite newbie with this plugin and need your help here
I'm trying to reproduce  [that](https://benenewton.medium.com/how-i-automatically-import-my-meetings-into-my-daily-note-template-3d9537b17dea) and get an error

app.js:1 Obsidian Developer Console
(unknown) Loading Hotkeys++ plugin
(unknown) loading url-into-selection
(unknown) Dataview: Version 0.4.26 Loaded
ObsidianOutlinerPlugin.ts:36 Loading obsidian-outliner
main.ts:40 loading Hotkeys for templates plugin v1.4.3
VM144:28331 Loading Markdown-Prettifier
VM145:760 Obsidian Admonition loaded
VM150:39 loading dictionary
main.ts:30 loading markdown-table-editor plugin
main.ts:20 Loading Mind Map plugin
main.ts:20 Loading natural language date parser plugin
main.ts:31 Loading Note Refactor plugin
main.ts:213 Recent Files: Loading plugin v1.3.2
main.ts:269 Recent Files: maxLength is not set, using default (50)
main.ts:18 loading plugin "tasks"
main.ts:48 loading obsidian-tracker plugin
VM160:15998 Loading QuickAdd
VM162:5387 Loading MetaEdit
VM162:5146 MetaEdit: (LOG) automator of type 'KanbanHelper' does not exist.
VM162:5146 MetaEdit: (LOG) automator of type 'ProgressProperties' does not exist.
main.ts:28 Loading Day Planner plugin
index.ts:154 Dataview: Task & metadata parsing queued in 0.064s.
main.ts:148 Hotkeys for templates -> active plugins: templater-obsidian,core
###    at Process.ChildProcess._handle.onexit (internal/child_process.js:267)
 ###    at onErrorNT (internal/child_process.js:469)
   ###  at processTicksAndRejections (internal/process/task_queues.js:84)**
log_error @ VM114:82
eval @ VM114:105
rejected @ VM114:57
Promise.then (async)
step @ VM114:62
eval @ VM114:63
__async @ VM114:47
errorWrapper @ VM114:98
eval @ VM114:3766
eval @ VM114:63
__async @ VM114:47
append_template_to_active_file @ VM114:3759
onChooseItem @ VM114:2361
t.onChooseSuggestion @ app.js:1
t.selectSuggestion @ app.js:1
e.useSelectedItem @ app.js:1
e.onSuggestionClick @ app.js:1
s @ app.js:1

Is someone can help here ?
Thanks 
Patrick

**Comment by shabegom** [comment]
Hi! It looks like there is a permissions error when trying to access `/usr/local/bin` to run `iCalBuddy`.

Are you comfortable with the command line? If not, this is going to be a bit challenging to troubleshoot.

First thing to try is specifying the full path to `iCalBuddy`

In the terminal run `which iCalBuddy`. It should return the full path (if it doesn't that mean iCalBuddy was not installed properly). Copy that full path and replace `iCalBuddy` in the User Template command with it. See if that changes your error.

There are multiple other things that could be causing this error, but I hope it's as simple as this.

**Comment by dgurkaynak** [comment]
I'm trying to do the exact same thing. As @shabegom suggested, you need to call `icalBuddy` by it's absolute path. It should look like something like this:
<img width="733" alt="image" src="https://user-images.githubusercontent.com/1475786/159336477-691fc845-68d8-4143-b02c-02396307270f.png">


However, I have another problem that I'm unable to solve for couple of days. When I insert this template, `icalBuddy` cannot find any calendars. I can see it in Obsidian console:
<img width="534" alt="image" src="https://user-images.githubusercontent.com/1475786/159337809-4d1c13a1-bcc6-43e4-b80f-e9cc4d6165a1.png">


The exact same command works in my terminal (iTerm). However the first time I ran it, macOS prompted me about permission stuff. After allowing iTerm to access my calendar, `icalBuddy` returns events as I wanted. But, macOS doesn't prompted me about permission stuff for Obsidian. In macOS System Preferences > Security & Privacy > Privacy > Calendar tab, I see the apps that I allowed before and there is no button to manually add an app :/
<img width="680" alt="image" src="https://user-images.githubusercontent.com/1475786/159337516-92ffbe08-34f3-4e90-9880-be0f01afd536.png">


Has anyone encountered this kind of permission issue before?

**Comment by pillouz01** [comment]
Hi
Thanks a lot @shabegom 
I have tried adding the full path but same issue
I have installed icalbuddy with hombrew
In terminal the command works perfectly 
Any other idea?
Thanks 
Patrick

**Comment by shabegom** [comment]
@dgurkaynak that seems weird that the container app needs permission and not the iCalBuddy script itself. I know with some of those screens you can drag/drop apps on the list to add them. 

@pillouz01 ah homebrew. Ok. 

So a couple of common things that break shell scripts:

1. Specify the full path to the shell you want to use in the templater settings (zsh, bash, etc)
2. The most common problem is that the script needs access to your $PATH which for some reason the child process doesn't read properly (see #512). The fix is to explicitly set the $PATH before running the command (something like `PATH=$PATH:/usr/local/bin`). 

If that doesn't work, maybe it is really just a permissions issue and you need to `chmod` the directory throwing the error? Seems odd that your user wouldn't have permissions tho.

**Comment by pillouz01** [comment]
Hi
Thanks a lot for the answer
I took another approach in that case

Copy the fixed icalbudy binary in a folder and create my custom action

It works fine like that for me

Thanks a lot !!

Regards
Patrick
